### PR TITLE
Tidy up some (rarely touched) tests #4235

### LINKED ISSUE #4235: Tidy up some (rarely touched) tests
Quite a number of mess in the test package:
- `BaseTestCaseTest` tests nothing but `AssertHelper`, should be renamed.
- `PageDataTest` tests `StringHelper.truncate`, and is never called.
- `test.cases.pageobjects` doesn't exist but is written on `testng.xml`.
- `ActivityLogTest` placed in `test.cases.ui`, but `ActivityLogEntry` is a util.
- `test.automated` contains test cases. All test cases should be placed in `test.cases.*`.

After resolving all, #4183 should be easier.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #4235 
How (a portion of) the test package looks like now:
test_package
And how (again, just a portion of) `testng.xml` looks now:
test_testng
Much neater IMO. Now all test cases are found only in `test.cases.*` package. In fact, all the test files match the regex `test\/cases\/[a-z]+(\/[a-z]+)*\/(?!\QBase\E)[A-Za-z]+\.java`, if anyone wants to test it that way.

### CODE REVIEW

**Comment by yj-soh** [comment]
Should the callback classes be moved out of their respective classes?
If the classes are exclusively for those they were in, it may be better to leave them there to denote that they aren't meant to be used elsewhere.

File: src/test/java/teammates/test/automated/FeedbackSessionsEmailTaskQueueCallback.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.HashMap;
+
+import teammates.common.util.Const;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.Const.ParamsNames;
+
+import com.google.appengine.api.urlfetch.URLFetchServicePb.URLFetchRequest;
+
+@SuppressWarnings("serial")
+public class FeedbackSessionsEmailTaskQueueCallback extends BaseTaskQueueCallback {
```
Comment (yj-soh) [inline src/test/java/teammates/test/automated/FeedbackSessionsEmailTaskQueueCallback.java]: Either include or eliminate the `TaskQueue` part in all of these classes?
I see you eliminated it from `SubmissionsAdjustmentTaskQueueCallback`, but it's still present here.

**Comment by wkurniawan07** [comment]
I'm personally not into nested classes, unless they're really needed (like for encapsulation purpose and stuff).
But then again, it's true that they're not used anywhere else. Guess I'll just put them back in and make them private (they were public before).

**Comment by wkurniawan07** [comment]
I reverted the renames. I realized that if I renamed them the way I wanted it to be (the callback classes' names follow the `*` in `*Action` and the test classes' names follow the `*` in `*Servlet`, as per the `logic.automated` package), the PR will be too big. Not that important anyway.

**Pull request merged**